frontend/src/utils/DeviceStore/deviceStore.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Device store that retrieves the stored device ID for a given device type (audio or video)
+ * and checks if it is still connected.
+ */
+export default class DeviceStore {
+  private devices: MediaDeviceInfo[] = [];
+
+  async init() {
+    if (navigator.mediaDevices) {
+      this.devices = await navigator.mediaDevices.enumerateDevices();
```

Comment:
I'm assuming this code flow still works  ok when we don't have user permissions yet